hedera-node/src/main/java/com/hedera/services/throttling/DeterministicThrottling.java

```diff
@@ -71,10 +76,18 @@ public boolean shouldThrottleQuery(HederaFunctionality queryFunction) {
 	public boolean shouldThrottleTxn(TxnAccessor accessor, Instant now) {
 		final var function = accessor.getFunction();
 		ThrottleReqsManager manager;
+
+		if ((function == ContractCreate || function == ContractCall) && dynamicProperties.shouldThrottleByGas()) {
+			final var txGasLimit = function == ContractCreate ?
+					accessor.getTxn().getContractCreateInstance().getGas() : accessor.getTxn().getContractCall().getGas();
+			if (basedOnGasThrottle == null || !basedOnGasThrottle.allow(now, txGasLimit)) {
```

Comment:
Should we immediately return `false` if this this condition fails? Otherwise we will "drop through" to the later tests and (it seems) either return the wrong answer and/or impact throttling of other operations.